Proposal changes got p2p homework CONTRIBUTING.md (#8)
kottans/frontend-2019-p2p: CONTRIBUTING.md

[![MIT Licensed][icon-mit]][license]

> 

Reviewer: Normally badges come in a single line. Let's keep it consistent across kottans repos.

  as previous section requires. You may find this
  [Re-sync Fork With Upstream guide](https://gist.github.com/OleksiyRudenko/8b3ddb664308de0634b53c525e551d8b)
  helpful to fix the issue.

### Codereview
- you will **require** approvals* from **at least two peers**, so seek
for sufficient support. 
  - ask your peers for review in [FE Students chat](https://web.telegram.org/#/im?p=s1495296464_549300875144563897),
  post a link to your PR as well
  - if in some reason there are not reviwers from p2p course, you can ask your friends outside of the course<br>

> - **you should provide a review** for this same task to **at least 2 PR's** of other students of p2p course

Reviewer: IMHO, this point is still discussable. The main question is whether a student is able to suggest improvements to a task the student has not actually completed yet?
My assumption is that a student that have a PR re certain task approved can bring new learnings further to other students, which means that a student can provide quality feedback to those who are completing an earlier tasks only.

Putting it other way round the suggestion is
```suggestion
- **you have to provide reviews** on the same task to **at least 2 PRs** of other students of p2p course once your PR is merged, using your learnings from merged PR code review
```

Hope the idea behind is clear.

> - after finishing previous steps ask mentors for review in [FE Questionarium chat](https://web.telegram.org/#/im?p=s1495296464_549300875144563897)

Reviewer: Is it about a **real** review or we just want someone to merge the PR?

Reviewer: @Oodmincheg Please, reflect on the comment above.

Author: I thought that minimenotors will do some reviews, and sometimes mentors

Reviewer: Clear, thanks. This gonna happen at the beginning for sure. We just do not want this on a regular basis, only from time to time.
What do we definitely need mentors for is merging. So student
- can summon a mentor when help is needed (e.g. the contributing student cannot decide on how to implement changes, or student reviewers see the issue in the code but cannot decide on proper way to fix it)
- has to summon a mentor in order to merge a pr

Not sure if item two should be covered in this section.

Please, suggest wordings for the above.

PS Please, do not use term *minimentor* as it sounds a bit underevaluing ) I'd suggest *junior mentor*, which is also questionable but at least it uses commonly accepted in the industry grade denomination.

> - once the code review phase successfully finished and you have approval from other students, post a message in [FE Students chat](https://web.telegram.org/#/im?p=s1495296464_549300875144563897): { Here must be the name of your task } — #p2p-pr-done and add the link to your PR. This step is important, as it helps mentors to track your progress!

Reviewer: ```suggestion
- once the code review phase successfully finished and you have approval from other students, post a message in [FE Students chat](https://web.telegram.org/#/im?p=s1495296464_549300875144563897): `{ Here must be the name of your task } — #p2p-pr-done` and add the link to your PR. This step is important, as it helps mentors to track your progress!
```

Just to have quotation styled a bit different from surrounding text.

Also note that MD renderers stick adjacent lines in `.md` files into a contiguous paragraph, while having long sentences and paragraphs split with a newline somewhere at 80 characters marks lesser text as changed.

> *Code reviews that are only about code style/code formatting, variables and function naming, introducing semantic constants do not count.*

Reviewer 2: `Code reviews that are only about code style/code formatting, variables and function naming, introducing semantic constants do not count.` I prefer to drop this line, because... well, sometimes there is nothing more to change in PR.

Reviewer: But if there is this is a clear way to cheat around )
Let them ask us if there is really nothing to do.

  situation and fix as appropriately. Do not expect code reviewers
  to attract your attention to every single line of code that
  requires similar fix. Please, take care of your peers for reciprocity
- implement changes in your **task dedicated repo** and **update demo**
  and commit
- copy changed files to this repo fork feature branch and commit
- push feature branch to your remote p2p repo to update your PR:
  `git push origin dom-api-task`
  (use actual feature branch name)
- supplement commits with messages that give an idea of what's inside
  of the commit without looking into it.
  [How to Write a Git Commit Message](https://chris.beams.io/posts/git-commit/)

**Go to your PR on github and let code reviewers know your are done:**

- **resolve conversations you believe you have elaborated on**
- make sure you answered questions and gave proper explanations where
  required (do not resolve those conversations until you have a feedback
  from a code reviewer)
- if any conversations remain unresolved let reviewers
  know your're done explicitly (e.g. write a comment mentioning
  a code reviewer using `@`)
  
### Merging the PR

> 

Reviewer: ```suggestion
PR can be merged by repo maintainers only. When you have required code reviews completed and all code reviewers have approved requested changes, just mention two or three mentors for them to merge approved PR into `master`.
```

  as previous section requires. You may find this
  [Re-sync Fork With Upstream guide](https://gist.github.com/OleksiyRudenko/8b3ddb664308de0634b53c525e551d8b)
  helpful to fix the issue.

### Codereview

> - you will **require** approvals\* from **at least two peers**, so seek

Reviewer: ```suggestion
- you will **require** approvals from **at least two peers**, so seek
```

than one commit. So it is important to let reviewers know
they should not expect any further commits. **C** is for care.

> 

Reviewer: I'd suggest keeping this section to have end-to-end process described. Even though there is a single line like "PR is merged by one of the mentors".

Reviewer: @Oodmincheg Please, reflect on the comment above.

Author: Add section back(if I understand well)
